src/dune_pkg/fetch.ml

```diff
@@ -135,10 +135,16 @@ let fetch ~unpack ~checksum ~target (url : OpamUrl.t) =
 ;;
 
 module Opam_repository = struct
-  type t = { url : OpamUrl.t }
+  type t = Workspace.Repository.t
 
-  let of_url url = { url }
-  let default = of_url @@ OpamUrl.of_string "https://opam.ocaml.org/index.tar.gz"
+  let of_url url =
+    let name = Workspace.Repository.Name.of_string "unknown" in
```

Comment:
What happen when multiple repos become "unknown"? Won't their download directories overlap?

Reply:
Yes, and it would create an issue. Currently the only place where this happens is when a repo is specified via command line, so maybe it would make sense to hash the path instead of using the name. Downside is that the repo names in the folder become strange hex character strings but it might be a better solution overall.

Reply:
I'm now building `name-digest(source)` directories, so this should avoid this case at the expense of having the same repo cached multiple time with the same name, but I don't think it is a big risk.